hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/TimelineService.java

```diff
@@ -342,8 +343,29 @@ private int startServiceOnPort(int port) throws IOException {
   }
 
   public int startService() throws IOException {
-    final Server server = timelineServerConf.numThreads == DEFAULT_NUM_THREADS ? new JettyServer(new JavalinConfig()).server() :
-            new Server(new QueuedThreadPool(timelineServerConf.numThreads));
+    int maxThreads;
+    if (timelineServerConf.numThreads > 0) {
+      maxThreads = timelineServerConf.numThreads;
+    } else {
+      // io.javalin.jetty.JettyUtil.defaultThreadPool
+      maxThreads = 250;
+    }
+    QueuedThreadPool pool = new QueuedThreadPool(maxThreads, 8, 60_000);
+    pool.setDaemon(true);
+    final Server server = new Server(pool);
```

Comment:
Thanks for the fix, is there any possibility we can write some tests for it?

Reply:
done